src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+    *   CALL :  MMMMXXXMMM   MMMM|FFFFFFF|MMMMMFMMMM|F|MMFFFMMM
+    *   TRUTH:  MMMM   MMMMMMMMMM-MMMMMMM|MMMMMMMMMM|F|MMMMMMMM
+    *             correct       long        short     long
+    *   CIGAR:  00003330002220000511111116000001000060600111000
+    * shows one known block split
+    */
+  def toCigar(pairedIterator: JointVariantContextIterator,
+              truthPhaseBlockDetector: OverlapDetectorByContig,
+              callPhaseBlockDetector: OverlapDetectorByContig,
+              metric: AssessPhasingMetric,
+              skipMismatchingAlleles: Boolean): Cigar = {
+    val cigar: Cigar = pairedIterator
+      .filter { // ensure the alleles are the same when both truth and call are called
+        case Seq(Some(t: VCtx), Some(c: VCtx)) =>
+          import scala.collection.JavaConversions.iterableAsScalaIterable
+          !skipMismatchingAlleles || t.getAlleles.toSet == c.getAlleles.toSet
+        case _                                 => true
+      }
+      .flatMap { case Seq(t, c) =>
+        val ctxs                 = Seq(t, c).flatten // NB: can be 1 or 2 contexts here
+        val inTruthPhaseBlock    = ctxs.exists(truthPhaseBlockDetector.overlapsAny)
+        val inCallPhaseBlock     = ctxs.exists(callPhaseBlockDetector.overlapsAny)
+        val isTruthVariant       = t.isDefined
+        val isCallVariant        = c.isDefined
+        val truthPhasedVariant   = t.filter { ctx => PhasingUtils.getPhasingSetId(ctx) > 0 }
+        val callPhasedVariant    = c.filter { ctx => PhasingUtils.getPhasingSetId(ctx) > 0 }
+        val isTruthVariantPhased = truthPhasedVariant.isDefined
+        val isCallVariantPhased  = callPhasedVariant.isDefined
+
+        // DEBUG: make sure we overlap at most one phase block
+        require(ctxs.forall(truthPhaseBlockDetector.getOverlaps(_).size <= 1))
+        require(ctxs.forall(callPhaseBlockDetector.getOverlaps(_).size <= 1))
+
+        // Fill in the metrics
+        // Basic metrics
+        if (isCallVariant) {
+          metric.num_called += 1
+          if (isCallVariantPhased) metric.num_phased += 1
+        }
+        if (isTruthVariant) {
+          metric.num_truth += 1
+          if (isTruthVariantPhased) metric.num_truth_phased += 1
+        }
+        // Less fun to compute metrics
+        if (isCallVariant && isTruthVariantPhased) {
+          metric.num_called_with_truth_phased += 1
+          if (isCallVariantPhased) metric.num_phased_with_truth_phased += 1
+        }
+        if (inCallPhaseBlock) {
+          if (isTruthVariantPhased) {
+            metric.num_truth_phased_in_called_block += 1
+            if (isCallVariantPhased) metric.num_both_phased_in_called_block += 1
+          }
+        }
+
+        // NB: only create the cigar from phased variants.
+        if (isCallVariantPhased || isTruthVariantPhased) {
+          contextsToCigar(truthPhasedVariant, callPhasedVariant)
+        }
+        else Seq.empty
+      }.toList match { // remove leading `*End`
```

Comment:
Comment is now ` remove leading `*End` since we are going to prepend Seq(BothEnd)`